vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/network/externalipranger/externalip_admission.go

```diff
@@ -155,6 +171,7 @@ func (r *externalIPRanger) Validate(a admission.Attributes, _ admission.ObjectIn
 			ip := net.ParseIP(s)
 			if ip == nil {
 				errs = append(errs, field.Forbidden(field.NewPath("spec", "externalIPs").Index(i), "externalIPs must be a valid address"))
+				invalidIp = true
```

Comment:
Kubernetes [already validates](https://github.com/openshift/origin/blob/17f1107c/vendor/k8s.io/kubernetes/pkg/apis/core/validation/validation.go#L3712) that all `service.Spec.ExternalIPs` values are valid IPs, so we don't need to check that here. So we don't need to distinguish admin-overridable from non-admin-overridable errors.

Reply:
Isn't it better to validate them early in the cycle before it gets to the spec level validation?

Reply:
validation should happen before admission controllers; there's no point in doing admission control on an object that will be rejected anyway